File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PaginationParams
    {

        public string CurrentUsername { get; set; }
        public string Gender { get; set; }
        public string PetType { get; set; }
        public int MinAge { get; set; } = 0;
        public int MaxAge { get; set; } = 30;
        public string OrderBy { get; set; } = "lastActive";
              
    }
}

File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    gender: string;
    petType: string;
    minAge = 0;
    maxAge = 30;
    pageNumber = 1;
    pageSize = 5;
    orderBy = 'lastActive';
 
    constructor(user: User) {
        this.gender = "";
        this.petType = user.petType === 'dog' ? 'dog' : 'cat';
    }
}